src/installer.rs

```diff
@@ -1,15 +1,20 @@
-use std::collections::HashMap;
 use std::fmt::Display;
-use std::fs::File;
-use std::path::PathBuf;
+use std::{
+    fs::{create_dir_all, remove_dir_all, File, OpenOptions},
+    io::{copy, Seek},
+    path::PathBuf,
+};
```

Comment:
wait oh no, i just realized that you used some trick to omit `std::<module>::` from methods; This will get confusing fast, so, yeah, I would personally revert it myself

Reply:
I prefer having `std` specified too; I'm not sure if this is a common Rust thing, but yeah, on the Java projects here? We prefer to have these things made explicit in order to prevent confusion and potential problems

Reply:
In Rust, to my knowledge, it's mostly directly imported. I have never seen code that specifies `std`, at most the module will be referenced (like I mentioned)

Reply:
It's certainly down to preference but this practice of directly importing functions and structs/traits is very common. There [doesn't seem to be a convention](https://github.com/rust-lang/fmt-rfcs/blob/master/guide/items.md#imports-use-statements) on when to use `use` declarations besides [to avoid wildcard imports](https://www.lurklurk.org/effective-rust/wildcard.html) which isn't an issue here.

Reply:
I like rust, because stuff is scoped to allow you to use shorter names when the scope is otherwise clear.
However, in certain situations using the scope greatly helps.

For your examples:
`File` is already plenty obvious. The only files we are interacting with are clearly filesystem ones.
`copy` can mean a lot of things. copying files? bytes? just a value? Therefore I would prefer scoping that with *at least* `io::copy`. I personally always use `std::io::copy` for these things, but I don't care too much.
Stuff like `create_dir_all` is fairly self-contained, so I wouldn't require scoping on it.

*Personally* I always provide at least *some* scope for free functions, to indicate that they are imported from somewhere.

Reply:
> provide at least some scope for free functions, to indicate that they are imported from somewhere.

That's what I've done at the moment. Let me know of what you guys think, and feel free to suggest commits on the importing style.